projects/vdk-plugins/vdk-jupyter/.plugin-ci.yml

```diff
@@ -8,29 +8,71 @@
       - always
 
 
-build-vdk-jupyterlab-extension:
-  stage: build
+.common-setup-vdk-jupyterlab-extension:
   image: "python:3.8-alpine"
   before_script:
     - cd projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension || exit 1
     - pip install -U pip setuptools pre-commit
-    - apk --no-cache add npm
-    - apk add rust cargo
-    - apk add libffi-dev
+    - apk --no-cache add npm rust cargo libffi-dev
     - pip install jupyter
+  retry: !reference [.retry, retry_options]
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
+      changes:
+        - "projects/vdk-plugins/vdk-jupyter/**/*"
+
+
+build-vdk-jupyterlab-extension:
+  stage: build
+  extends: .common-setup-vdk-jupyterlab-extension
   script:
     - echo "Install Dependencies..."
     - npm ci
     - echo "Install Vdk JupyterLab Extension..."
     - pip install .
     - echo "Building VDK JupyterLab Extension..."
     - npm rebuild
-  retry: !reference [.retry, retry_options]
-  rules: # we want to trigger build jobs if there are changes to this package,
-    # but not if there are changes to VDK plugins or the main directory
-    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
-      changes:
-        - "projects/vdk-plugins/vdk-jupyter/**/*"
+
+
+server-tests-vdk-jupyterlab-extension:
+  extends: .common-setup-vdk-jupyterlab-extension
+  stage: test
```

Comment:
You may notice from the error https://gitlab.com/vmware-analytics/versatile-data-kit/-/pipelines/962699510

But there's not test stage. Let's use `build` stage.

Reply:
This would mean that we can add "stage: build" to the common setup, right?